### PR TITLE
Change CalendarButton to use <Tag />

### PR DESCRIPTION
**Pull request by elanoism**
**Changes**
* Rename ComponentButton to ComponentTag
* Fix tags so their size is always 32px and selected state has proper border (same as background)

Most of the lines are rename of the Button -> Tag rename (sorry about that).
Actual changes are mostly located in `src/components/Tag/Tag.js` and the corresponding story.

### CODE REVIEW

File: src/components/Tag/Tag.js [hunk 1]
```diff
@@ -52,13 +56,18 @@ const tagClickableStyles = ({ onClick, theme }) =>
 
     &:hover {
       background-color: ${theme.colors.n300};
+      ${shadowBorder(theme.colors.n300)};
     }
   `;
 
 const iconStyles = ({ theme }) => css`
   label: tag__icon;
   vertical-align: -2px;
```
Comment (fernandofleury) [inline src/components/Tag/Tag.js]: I'm not a big fan of magic numbers here. Can't we make it a variable then reuse it on the `svg inherit` declaration after it? (This way we can avoid the inherit at all)

File: src/components/CalendarTag/CalendarTag.js [hunk 2]
```diff
@@ -103,16 +103,16 @@ const CalendarButtonWrap = styled('div')`
...
-CalendarButton.propTypes = {
+CalendarTag.propTypes = {
   /**
    * Callback to receive the set of dates when the user selects them.
    */
   onDatesRangeChange: PropTypes.func.isRequired
 };
 
-CalendarButton.defaultProps = {};
+CalendarTag.defaultProps = {};
```
Comment (felixjung) [inline src/components/CalendarTag/CalendarTag.js]: If they are empty, feel free to remove them, I'd say.

File: src/components/CalendarTagTwoStep/CalendarTagTwoStep.js [hunk 3]
```diff
@@ -5,10 +5,10 @@ import styled, { css } from 'react-emotion';
 import { START_DATE, END_DATE } from 'react-dates/constants';
 
 import { RangePickerController } from '../Calendar';
-import Button from '../Button';
+import Tag from '../Tag';
 import { textKilo } from '../../styles/style-helpers';
 
-class CalendarButtonTwoStep extends Component {
+class CalendarTagTwoStep extends Component {
```
Comment (felixjung) [inline src/components/CalendarTagTwoStep/CalendarTagTwoStep.js]: I still find this name weird, but forgot what it actually does. 😞

File: src/components/CalendarTagTwoStep/CalendarTagTwoStep.js [hunk 4]
```diff
@@ -134,19 +134,19 @@ const baseStyles = () => css`
...
+const CalendarWrapper = styled.div`
   margin-top: ${({ theme }) => theme.spacings.byte};
 `;
 
-const CalendarButtonWrap = styled('div')`
+const CalendarTagWrapper = styled('div')`
   ${baseStyles};
 `;
 
 /**
```
Comment (felixjung) [inline src/components/CalendarTagTwoStep/CalendarTagTwoStep.js]: This should go just before your class definition further up.

File: src/components/CalendarTagTwoStep/CalendarTagTwoStep.js [hunk 5]
```diff
@@ -163,12 +163,12 @@ CalendarButtonTwoStep.propTypes = {
...
+CalendarTagTwoStep.defaultProps = {
   clearText: 'Clear',
   confirmText: 'Apply'
 };
 
 /**
  * @component
  */
-export default CalendarButtonTwoStep;
+export default CalendarTagTwoStep;
```
Comment (felixjung) [inline src/components/CalendarTagTwoStep/CalendarTagTwoStep.js]: For a class component, I feel it's totally fine to do `export default class CalendarTagTwoStep`.

File: src/components/Tag/Tag.js (hunk 1 above)
Comment (felixjung) [reply src/components/Tag/Tag.js]: Is this necessary because of the `display: inline-block`?

**Pull request merged**